etn-ccis/blui-react-workflows
# Enhance customization of error messaging

Fixes BLUI-4088.



Fixes BLUI-4401.



#### Changes proposed in this Pull Request:

- Add ErrorManager to remaining components
- Add useErrorManager hook
- Manage API errors via ErrorManager for all components that make API calls



#### Screenshots / Screen Recording (if applicable)

-



#### To Test:

- add 
``` 
throw new Error('My Custom Error', {
    cause: {
        title: 'Custom Title',
        errorMessage: 'My custom error message',
    },
});
```
to the beginning of the corresponding auth or registration action so that an error is thrown and the corresponding component that calls that action will render the appropriate error. By default the error message will be rendered inside of an error dialog. In order to test the message box simple change the props via the AuthContextProvider or the RegistrationContextProvider as follows:
```
 <AuthContextProvider
                    ...
                     errorConfig={{
                        mode: 'message-box',
                        messageBoxConfig: {
                            dismissible: true,
                            position: 'top',
                        },
                    }}
                >
```
If you provide no customization it should default to a dialog error. Feel free to check out the types definition and mess around with all of the props to ensure they all work.
